extensions/binding/org.eclipse.smarthome.binding.bluetooth/src/main/java/org/eclipse/smarthome/binding/bluetooth/BeaconBluetoothHandler.java

```diff
@@ -67,18 +70,23 @@ public void initialize() {
         }
 
         adapter = (BluetoothAdapter) bridgeHandler;
+
+        deviceLock.lock();
```

Comment:
locks should generally be guarded by `try {} finally {}`-blocks - especially when some (foreign) non-trivial code is called while the lock is in place.

Reply:
thx. updated.